scripts/install.sh

```diff
@@ -0,0 +1,165 @@
+            curl -fsSL https://deb.nodesource.com/setup_lts.x -o ${phala_scripts_tools_dir}/get-node.sh
+          fi
+          bash ${phala_scripts_tools_dir}/get-node.sh
+          apt-get install -y nodejs
+        ;;
+      esac
+    fi
+  done
+
+}
+
+function phala_scripts_install_sgx() {
+  _kernel_version=$(uname -r)
+  if [ "$1" == "uninstall" ];then
+    shift
+    phala_scripts_log info "Uninstall Install Sgx device" cut
+
+    # frist uninstall
+    # apt autoremove -y libsgx-enclave-common sgx-aesm-service
+    apt autoremove -y libsgx-enclave-common dkms
+    # [ -c /dev/sgx_enclave ]
+    [ -f /opt/intel/sgxdriver/uninstall.sh ] && bash /opt/intel/sgxdriver/uninstall.sh
+    # [[ "${_kernel_version}" =~ "5.4" ]] && apt autoremove -y intel-sgx-dkms
+    # if [[ "${_kernel_version}" =~ "5.4" ]];then
+      # [ -f /opt/intel/sgxdriver/uninstall.sh ] && bash /opt/intel/sgxdriver/uninstall.sh
+    # fi
+    return 0
+  fi
+  phala_scripts_log info "Kernel ${_kernel_version}" cut
+  phala_scripts_log info "Install Sgx device"
+  if [ -c /dev/sgx_enclave ];then
+    phala_scripts_install_sgx_default
+  elif [ ${DISTRIB_RELEASE} == "20.04" ];then
+    phala_scripts_install_sgx_k5_4 && \
+    phala_scripts_install_sgx_default
+  elif [ ${DISTRIB_RELEASE} == "18.04" ];then
+    phala_scripts_install_sgx_k5_4
+  fi
+
+  # if [[ "${_kernel_version}" =~ "5.13" ]];then
+  #   phala_scripts_install_sgx_default
+  # elif [[ "${_kernel_version}" =~ "5.4" ]];then
+  #   # first install
+  #   phala_scripts_install_sgx_k5_4 && \
+  #   phala_scripts_install_sgx_default
+  # else
+  #   return 1
+  # fi
+}
+
+function phala_scripts_install_sgx_default() {
+  # install aesm encalave
+  curl -fsSL https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add - && \
+  add-apt-repository -y "deb https://download.01.org/intel-sgx/sgx_repo/ubuntu focal main" && \
+  # reinstall : fix apt upgrade
+  # 21.10 sgx-aesm-service error skip aesm
+  # apt reinstall -y libsgx-enclave-common sgx-aesm-service
+  apt autoremove -y libsgx-enclave-common
+  apt install -y libsgx-enclave-common 
+
+}
+
+function phala_scripts_install_sgx_k5_4(){
+  # [ -f ${phala_scripts_tools_dir}/sgx_linux_x64_driver_2.11.0_2d2b795.bin ] || {
+    # curl -fsSL https://download.01.org/intel-sgx/latest/linux-latest/distro/ubuntu20.04-server/sgx_linux_x64_driver_2.11.0_2d2b795.bin -o ${phala_scripts_tmp_dir}/sgx_linux_x64_driver_2.11.0_2d2b795.bin && \
+    # mv ${phala_scripts_tmp_dir}/sgx_linux_x64_driver_2.11.0_2d2b795.bin ${phala_scripts_tools_dir}/
+  # }
+  # bash ${phala_scripts_tools_dir}/sgx_linux_x64_driver_2.11.0_2d2b795.bin
+
+  type make dkms >/dev/null 2>&1 || apt install -y make dkms
+  [ -f ${phala_scripts_tools_dir}/sgx_linux_x64_driver_1.41.bin ] || {
+    curl -fsSL https://download.01.org/intel-sgx/latest/linux-latest/distro/ubuntu20.04-server/sgx_linux_x64_driver_1.41.bin -o ${phala_scripts_tmp_dir}/sgx_linux_x64_driver_1.41.bin && \
```

Comment:
Hope we can extract URL like this to constant, then we shall easy to maintain them (e.g. upgrade for security reason)